qiskit/algorithms/evolvers/variational/var_qrte.py

```diff
@@ -0,0 +1,123 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Variational Quantum Real Time Evolution algorithm."""
+from typing import Optional, Union, Type, Callable, List, Dict
+
+import numpy as np
+from scipy.integrate import OdeSolver
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.evolvers.real_evolver import RealEvolver
+from qiskit.circuit import Parameter
+from qiskit.opflow import ExpectationBase, OperatorBase
+from qiskit.utils import QuantumInstance
+from . import RealMcLachlanPrinciple
+from .solvers.ode.forward_euler_solver import ForwardEulerSolver
+from .variational_principles import RealVariationalPrinciple
+from .var_qte import VarQTE
+
+
+class VarQRTE(VarQTE, RealEvolver):
+    """Variational Quantum Real Time Evolution algorithm.
+
+    .. code-block::python
+
+        from qiskit.algorithms import EvolutionProblem
+        from qiskit.algorithms import VarQITE
+        from qiskit import BasicAer
+        from qiskit.circuit.library import EfficientSU2
+        from qiskit.opflow import SummedOp, I, Z, Y, X
+        from qiskit.algorithms.evolvers.variational import (
+            RealMcLachlanPrinciple,
+        )
+        from qiskit.algorithms import EvolutionProblem
+        import numpy as np
+
+        observable = SummedOp(
+            [
+                0.2252 * (I ^ I),
+                0.5716 * (Z ^ Z),
+                0.3435 * (I ^ Z),
+                -0.4347 * (Z ^ I),
+                0.091 * (Y ^ Y),
+                0.091 * (X ^ X),
+            ]
+        ).reduce()
+
+        ansatz = EfficientSU2(observable.num_qubits, reps=1)
+        parameters = ansatz.parameters
+        init_param_values = np.zeros(len(ansatz.parameters))
+        for i in range(len(ansatz.parameters)):
+            init_param_values[i] = np.pi / 2
+        param_dict = dict(zip(parameters, init_param_values))
+        var_principle = RealMcLachlanPrinciple()
+        backend = BasicAer.get_backend("statevector_simulator")
+        time = 1
+        evolution_problem = EvolutionProblem(observable, time)
+        var_qrte = VarQRTE(ansatz, var_principle, param_dict, quantum_instance=backend)
+        evolution_result = var_qite.evolve(evolution_problem)
+    """
+
+    def __init__(
+        self,
+        ansatz: Union[OperatorBase, QuantumCircuit],
+        variational_principle: RealVariationalPrinciple = RealMcLachlanPrinciple(),
```

Comment:
Having what is mutable object as a default can up with inconsistent behavior since a single instance is maintained for this default. Hence if the instance changes the default changes. Better to default to None here and internally if None then create an instance. (Same comment for Imaginary change that is similar to this).